crates/autopilot/src/util/mod.rs

```diff
@@ -0,0 +1,14 @@
+use url::Url;
+
+/// Joins a path with a URL, ensuring that there is only one slash between them.
+/// It doesn't matter if the URL ends with a slash or the path starts with one.
+pub fn join(url: &Url, mut path: &str) -> Url {
+    let mut url = url.to_string();
+    while url.ends_with('/') {
+        url.pop();
+    }
+    while path.starts_with('/') {
+        path = &path[1..]
+    }
+    Url::parse(&format!("{url}/{path}")).unwrap()
+}
```

Comment:
copied from https://github.com/cowprotocol/services/blob/main/crates/shared/src/url.rs#L5 to break the dependency to `shared`. This is already done for `solvers` crate also: https://github.com/cowprotocol/services/blob/main/crates/solvers/src/util/url.rs

Reply:
https://github.com/cowprotocol/services/issues/2285